src/util/getjsonfromurl.ts

```diff
@@ -1,15 +1,14 @@
-function getjsonfromurl(id) {
-  return fetch(
-    "https://capstonewebclient.blob.core.windows.net/apparatus/" + id + ".json",
-    { mode: "cors" }
-  )
-    .then((response) => response.json())
-    .then((responseJson) => {
-      return responseJson;
-    })
-    .catch((error) => {
-      return error;
-    });
+async function getjsonfromurl(id: string) {
+  try {
+    const response = await fetch(
+      `https://capstonewebclient.blob.core.windows.net/apparatus/${id}.json`,
+      { mode: "cors" }
+    );
+    const responseJson = await response.json();
+    return responseJson;
```

Comment:
> > Works great, but has 1 minor code smell.
> 
> the code smell is an issue with the await which should be that way.

Changing **_const responseJson = await response.json();_** to **_return await response.json();_** should still be possible with awaits.